prover/circuits/execution/pi.go

```diff
@@ -85,145 +96,78 @@ func (l *L2MessageHashes) CheckSumMiMC(api frontend.API) frontend.Variable {
 		sumIsUsed = api.Add(sumIsUsed, isUsed)
 	}
 
-	api.AssertIsEqual(sumIsUsed, l.Length)
+	api.AssertIsEqual(sumIsUsed, s.Length)
 	return res
 }
 
 type FunctionalPublicInputSnark struct {
 	FunctionalPublicInputQSnark
-	InitialStateRootHash     frontend.Variable
-	InitialBlockNumber       frontend.Variable
-	InitialRollingHash       [32]frontend.Variable
-	InitialRollingHashNumber frontend.Variable
-	ChainID                  frontend.Variable
-	L2MessageServiceAddr     frontend.Variable
-}
-
-type FunctionalPublicInput struct {
-	DataChecksum             [32]byte
-	L2MessageHashes          [][32]byte
-	MaxNbL2MessageHashes     int
-	FinalStateRootHash       [32]byte
-	FinalBlockNumber         uint64
-	FinalBlockTimestamp      uint64
-	FinalRollingHash         [32]byte
-	FinalRollingHashNumber   uint64
-	InitialStateRootHash     [32]byte
-	InitialBlockNumber       uint64
-	InitialBlockTimestamp    uint64
-	InitialRollingHash       [32]byte
-	InitialRollingHashNumber uint64
-	ChainID                  uint64
-	L2MessageServiceAddr     types.EthAddress
+	InitialStateRootHash frontend.Variable
+	InitialBlockNumber   frontend.Variable
+	ChainID              frontend.Variable
+	L2MessageServiceAddr frontend.Variable
 }
 
 // RangeCheck checks that values are within range
-func (pi *FunctionalPublicInputQSnark) RangeCheck(api frontend.API) {
+func (spiq *FunctionalPublicInputQSnark) RangeCheck(api frontend.API) {
 	// the length of the l2msg slice is range checked in Concat; no need to do it here; TODO do it here instead
 	rc := rangecheck.New(api)
-	for _, v := range pi.L2MessageHashes.Values {
+	for _, v := range spiq.L2MessageHashes.Values {
 		for i := range v {
 			rc.Check(v[i], 8)
 		}
 	}
-	for i := range pi.FinalRollingHash {
-		rc.Check(pi.FinalRollingHash[i], 8)
+	for i := range spiq.FinalRollingHashUpdate {
+		rc.Check(spiq.FinalRollingHashUpdate[i], 8)
 	}
+	rc.Check(spiq.FinalBlockNumber, 64)
```

Comment:
I am fine leaving theses constraints, but just know that this is range-checked by the wizard already. We can leave it for the sake of carefulness.

Reply:
We can keep in mind that we can drop these checks if the PI circuit ever became a bottleneck.